src/openrct2/park/ParkFile.cpp

```diff
@@ -1193,7 +1193,12 @@ namespace OpenRCT2
                     // Ride ID
                     cs.ReadWrite(rideId);
 
-                    auto& ride = *RideAllocateAtIndex(rideId);
+                    auto& ride = [&]() -> Ride& {
+                        if (cs.GetMode() == OrcaStream::Mode::READING)
+                            return *GetRide(rideId);
+                        else
+                            return *RideAllocateAtIndex(rideId);
```

Comment:
This looks like it’s the wrong way round - it uses `RideAllocateAtIndex` when _saving_.

Reply:
Oh yeah